alphabase/io/tempmmap.py

```diff
@@ -22,6 +22,73 @@
     "and might need to be triggered manually!"
 )
 
+def _check_temp_dir_existence(abs_path: str) -> str:
+    """
+    Check if the directory to which the temp arrays should be written exists. If not raise a value error.
+
+    Parameters
+    ----------
+    abs_path : str
+        The absolute path to the new temporary directory.
+
+    Returns
+    ------
+    str 
+        The directory path if it exists.
+    """
+    #ensure that the path exists
+    if os.path.exists(abs_path):
+        # ensure that the path points to a directory
+        if os.path.isdir(abs_path):  
+            TEMP_DIR_NAME = abs_path
```

Comment:
I think this is a decisive side effect that should be reflected in both the method name and the docstring ;-)

Reply:
e.g. rename to `_set_new_temp_dir`

Reply:
I don't think I understand what you mean with decisive side effect? This function does not actually do anything except raise a value error if the provided path does not fulfil the requirements (here that it exists and that it points to a directory).

Reply:
it does something very fundamental: it changes `TEMP_DIR_NAME`, thus altering the state of the module (which is exactly what we want at this point, so all fine, just the name should reflect that)